receiver/prometheusreceiver/metrics_receiver_honor_timestamp_test.go

```diff
@@ -0,0 +1,550 @@
+				{code: 200, data: honorTimestampsPage1},
+				{code: 500, data: ""},
+				{code: 200, data: honorTimestampsPage2},
+			},
+			// since honortimestamp is set to false, so the resulted metrics should be same as metrics of 'TestCoreMetricsEndtoEnd' in metrics_receiver_test.go
+			validateFunc: verifyHonorTimeStampsFalse,
+		},
+	}
+	pConfig := &promConfig{
+		honorLabel: false,
+	}
+	testComponent(t, targets, pConfig, false, "")
+}
+
+func setMetricsTimestamp() {
+	onlyOnce.Do(func() {
+		honorTimestampsPage1 = fmt.Sprintf(honorTimestampsPage1,
+			ts1,      //timestamp for gauge
+			ts2, ts3, //timestamp for counter
+			ts4, ts4, ts4, ts4, ts4, ts4, //timestamp for histogram
+			ts5, ts5, ts5, ts5, ts5, //timestamp for summary
+		)
+		honorTimestampsPage2 = fmt.Sprintf(honorTimestampsPage2,
+			ts6,      //timestamp for gauge
+			ts7, ts8, //timestamp for counter
+			ts9, ts9, ts9, ts9, ts9, ts9, //timestamp for histogram
+			ts10, ts10, ts10, ts10, ts10, //timestamp for summary
+		)
+		honorTimestampsPage3 = fmt.Sprintf(honorTimestampsPage3,
+			ts11,       //timestamp for gauge
+			ts12, ts13, //timestamp for counter
+			ts14, ts14, ts14, ts14, ts14, ts14, //timestamp for histogram
+			ts15, ts15, ts15, ts15, ts15, //timestamp for summary
+		)
+	})
+}
+
+func verifyHonorTimeStampsTrue(t *testing.T, td *testData, resourceMetrics []*pdata.ResourceMetrics) {
+	verifyNumScrapeResults(t, td, resourceMetrics)
+	m1 := resourceMetrics[0]
+	// m1 has 4 metrics + 5 internal scraper metrics
+	assert.Equal(t, 9, metricsCount(m1))
+
+	wantAttributes := td.attributes
+	e1 := []testExpectation{
+		assertMetricPresent("go_threads",
+			compareMetricType(pdata.MetricDataTypeGauge),
+			[]dataPointExpectation{
+				{
+					numberPointComparator: []numberPointComparator{
+						compareTimestamp(pdata.NewTimestampFromTime(time.UnixMilli(ts1))),
+						compareDoubleValue(19),
+					},
+				},
+			}),
+		assertMetricPresent("http_requests_total",
+			compareMetricType(pdata.MetricDataTypeSum),
+			[]dataPointExpectation{
+				{
+					numberPointComparator: []numberPointComparator{
+						// For initial scrape, which does not have prior scrape and hence prior explicit timestamp:
+						// Start_timestamp is same as point timestamp, read from the explicit timestamp given to the datapoint in the testdata.
+						//compareStartTimestamp(pdata.NewTimestampFromTime(time.UnixMilli(ts2))),
+						compareTimestamp(pdata.NewTimestampFromTime(time.UnixMilli(ts2))),
+						compareDoubleValue(100),
+						compareAttributes(map[string]string{"method": "post", "code": "200"}),
+					},
+				},
+				{
+					numberPointComparator: []numberPointComparator{
+						// For initial scrape, which does not have prior scrape and hence prior explicit timestamp:
+						// Start_timestamp is same as point timestamp, read from the explicit timestamp given to the datapoint in the testdata.
+						//compareStartTimestamp(pdata.NewTimestampFromTime(time.UnixMilli(ts3))),
```

Comment:
Same here.

Reply:
@Aneurysm9 I have commented it out because counter metrics timestamp is not coming correct. 
Currently, counter metrics `start_timestamp` is always equal to Gauge metric explicit timestamp in every scrape- which does not pass the validation. I have updated the comments with TODO remark.

Reply:
@Aneurysm9 Thanks. removed missing scrapes.